src/libs/Network/SequentialQueue.js

```diff
@@ -26,18 +28,41 @@ let currentRequest = null;
 function process() {
     const persistedRequests = PersistedRequests.getAll();
 
-    // This sanity check is also a recursion exit point
-    if (NetworkStore.isOffline() || _.isEmpty(persistedRequests)) {
+    // If we have no persisted requests or we are offline we don't want to make any requests so we return early
+    if (_.isEmpty(persistedRequests) || NetworkStore.isOffline()) {
+        isSequentialQueueRunning = false;
         return Promise.resolve();
     }
 
-    const task = _.reduce(persistedRequests, (previousRequest, request) => previousRequest.then(() => {
-        currentRequest = Request.processWithMiddleware(request, true);
-        return currentRequest;
-    }), Promise.resolve());
+    isSequentialQueueRunning = true;
```

Comment:
Can you explain why we are setting `isSequentialQueueRunning` here too? Isn't this handle in `flush()`? I think we can improve (DRY) the code here.

Reply:
I think you're right that it's redundant. We only call process from flush and flush always sets it to true first. I'll remove it. Thanks!

Reply:
I think you forgot another redundant line `isSequentialQueueRunning = false;` inside `process`. flush will also set this to false on `finally`.